### ISSUE TITLE
varlib instancer fails on Version 1.0 STAT tables

### ISSUE THREAD

**Issue by aaronbell** [fonttools/fonttools#2933]
I have a varLib.instancer script that I'm running to produce instances on a variety of fonts, but found that the script fails to updateFontNames when the STAT table is set to version 1.0, specifically due to the lack of the ElidedFallbackNameID field. 

I tracked the issue down to names.py where
```
        if not all(getName(i, *platform) for i in (1, 2, elidedNameID)):
            # Since no family name and subfamily name records were found,
            # we cannot update this set of name Records.
            continue
```
When ElidedFallbackNameID is missing, a elidedNameID value of 30564 is assigned, which is (obviously) not found in the font. Therefore, the above check fails and the font names are not updated. 

I've "fixed" the issue by inserting a check for this abnormal elidedNameID value into line 207 that defaults it to use the font's subFamilyName:

```
    axisValueNameIDs = [a.ValueNameID for a in axisValues]
    ribbiNameIDs = [n for n in axisValueNameIDs if _isRibbi(nametable, n)]
    nonRibbiNameIDs = [n for n in axisValueNameIDs if n not in ribbiNameIDs]
    elidedNameID = stat.ElidedFallbackNameID
    elidedNameIsRibbi = _isRibbi(nametable, elidedNameID)

    if elidedNameID == 30564:
        elidedNameID = 2
```

With this correction, the font instances export and the individual instances have corrected font names. While Version 1.0 STAT tables are not very common, it would be good to have a check in the script to make sure this doesn't happen.

**Comment by justvanrossum** [comment]
If I undersand correctly, for STAT 1.0 tables ,`stat.ElidedFallbackNameID` should be `None`, so I'm not sure where this 30564 number could come from.

That said, the check you quote should also fail if `elidedNameID` is `None`, so perhaps there are two problems here.

**Comment by aaronbell** [comment]
Yes, I expected it would be `None` too, and was surprised to see instead 30564.

I’m not totally sure what is going on with it but figured I’d raise the issue. Was just glad to be able to move forward on my little project!
